app/models/program.rb

```diff
@@ -150,7 +150,7 @@ def languages_list
   # * +False+ -> If there is not any event for the given date
   def any_event_for_this_date?(date)
     parsed_date = DateTime.parse("#{date} 00:00").utc
-    events.where(start_time: parsed_date..(parsed_date + 1)).any?
+    EventSchedule.where(schedule_id: selected_schedule.id).where(start_time: parsed_date..(parsed_date + 1)).any?
```

Comment:
This is great! :+1: 

Just one thing, you can use the schedule directly without using the `id`, which is more readable and it is using the Rails relation instead of the `id` column. I mean: 

`EventSchedule.where(schedule: selected_schedule).where(start_time: parsed_date..(parsed_date + 1)).any?`